components/patient-data-sharing/push-api/src/test/java/org/phenotips/data/push/internal/DefaultPushServerSendPatientResponseTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.data.push.internal;
+
+import net.sf.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class DefaultPushServerSendPatientResponseTest {
+
+    private DefaultPushServerSendPatientResponse pushResponse;
+
+    @Before
+    public void setUp()
+    {
+        JSONObject jsonObject = new JSONObject();
+        pushResponse = new DefaultPushServerSendPatientResponse(jsonObject);
+    }
+
+    @Test
+    public void isActionFailed_incorrectGroupVerifiesTrueKey()
+    {
+        pushResponse.response.accumulate("incorrect_user_group", true);
+        Assert.assertTrue(pushResponse.isActionFailed_incorrectGroup());
+    }
+
+    @Test
+    public void isActionFailed_UpdateDisabledVerifiesTrueKey()
+    {
+        pushResponse.response.accumulate("updates_disabled", true);
+        Assert.assertTrue(pushResponse.isActionFailed_UpdatesDisabled());
+    }
+
+    @Test
+    public void isActionFailed_IncorrectGUIDVerifiesTrueKey() {
+        pushResponse.response.accumulate("incorrect_guid", true);
+        Assert.assertTrue(pushResponse.isActionFailed_IncorrectGUID());
+    }
+
+    @Test
+    public void isActionFailed_GUIDAccessDeniedVerifiesTrueKey() {
+        pushResponse.response.accumulate("guid_access_denied", true);
+        Assert.assertTrue(pushResponse.isActionFailed_GUIDAccessDenied());
```

Comment:
A test should also check the other state. Set the key to false, check the response is now false.